From 32f7cb9e1399d7bd14f3642649329893ca7b1d51 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 15 Aug 2026 23:11:43 +0100
Subject: [PATCH 1/1] data/macros: Update bolt optimization flags

Replace deprecated hfsort+ with cdsort

Replace deprecated -icf=1 usage with -icf

Remove -use-gnu-stack, no longer required since binutils 2.44
https://forge.sourceware.org/binutils-gdb/binutils-gdb-mirror/commit/bbc969306f8d5fb6c7b636e25f6f8e278946ef23
---
 data/macros/actions/pgo.yaml | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/data/macros/actions/pgo.yaml b/data/macros/actions/pgo.yaml
index 5a76ee6..4fc318f 100644
--- a/data/macros/actions/pgo.yaml
+++ b/data/macros/actions/pgo.yaml
@@ -31,13 +31,12 @@ actions:
         boptim(){
             llvm-bolt ${1}.orig -o ${1}.bolt -data=$PKG_BUILD_DIR/BOLT/final/$(basename ${1}).fdata \
                 -reorder-blocks=ext-tsp \
-                -reorder-functions=hfsort+ \
+                -reorder-functions=cdsort \
                 -split-functions \
                 -split-all-cold \
                 -split-eh \
                 -dyno-stats \
-                -icf=1 \
-                -use-gnu-stack \
+                -icf \
                 -peepholes=all \
                 -frame-opt=hot \
                 -use-old-text ${2} ${3} ${4}
-- 
2.55.0

